src/devices/machine/mm5740.cpp

```diff
@@ -138,26 +139,39 @@ TIMER_CALLBACK_MEMBER(mm5740_device::perform_scan)
 					}
 				}
 			}
-			else    // key released, unmark it from the "down" info
+			else
 			{
+				// key released, unmark it from the "down" info
 				m_x_mask[x] &= ~(1 << y);
 			}
```

Comment:
You need to test here whether the released key is the most recently detected key-down, and if it is, you should stop generating additional repeat events.